test/experimental/test_vocab.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+from collections import OrderedDict
+import os
+import torch
+
+from test.common.torchtext_test_case import TorchtextTestCase
+from torchtext.experimental.vocab import (
+    Vocab
+)
+
+
+class TestVocab(TorchtextTestCase):
+    def tearDown(self):
+        super().tearDown()
+        torch._C._jit_clear_class_registry()
+        torch.jit._recursive.concrete_type_store = torch.jit._recursive.ConcreteTypeStore()
+
+    def test_has_unk(self):
+        c = OrderedDict({})
+        v = Vocab(c)
+
+        # check if unk is mapped to the first index
+        self.assertEqual(v['not_in_it'], 0)
+        self.assertEqual(v['<unk>'], 0)
+
+    def test_new_unk(self):
+        c = OrderedDict({})
+        v = Vocab(c, unk_token="<new_unk>")
+
+        # check if new_unk is mapped to the first index
+        self.assertEqual(v['<new_unk>'], 0)
+        self.assertEqual(v['not_in_it'], 0)
+
+    def test_vocab_get_item(self):
+        token_to_freq = {'a': 2, 'b': 2}
+        sorted_by_freq_tuples = sorted(token_to_freq.items(), key=lambda x: x[1], reverse=True)
+        c = OrderedDict(sorted_by_freq_tuples)
+
+        v = Vocab(c, specials=['<pad>', '<eos>'])
+        self.assertEqual(v['<unk>'], 0)
+        self.assertEqual(v['<pad>'], 1)
+        self.assertEqual(v['<eos>'], 2)
+        self.assertEqual(v['a'], 3)
+        self.assertEqual(v['b'], 4)
+
+    def test_vocab_set_item(self):
```

Comment:
I took a look at torchtext.Vocab and there are no similar func. There is also no such func in [pytext/ScriptVocabulary](https://github.com/facebookresearch/pytext/blob/master/pytext/torchscript/vocab.py#L11).

I think a reasonable scenario is that users can always build a new vocab with the custom order or new tokens. It might be too complicate for us to maintain such capability.

Reply:
I see so you're suggesting to not keep the functionality to allow users to update an index once the vocab has been created?

Reply:
I think it's not necessary. Do you see this kind of func in other vocab?